crates/core/tedge_api/src/entity_store.rs

```diff
@@ -308,13 +313,14 @@ impl EntityMetadata {
     }
 
     /// Creates a entity metadata for a child device.
-    pub fn child_device(child_device_id: String) -> Self {
-        Self {
+    pub fn child_device(child_device_id: String) -> Result<Self, TopicIdError> {
+        Ok(Self {
+            topic_id: EntityTopicId::default_child_device(&child_device_id)?,
```

Comment:
`default_child_device` is a single `format!()` call and can't return any errors

Reply:
If the given `child_device_id` is something containing a slash like `foo/bar`, I think `default_child_device` will return an error.

Reply:
@rina23q is correct. Using `"foo/bar"` as a child-device id is detected as an error by `impl FromStr for EntityTopicId`.